### PR TITLE
Support granular package data

### PR DESCRIPTION
**Pull request by semtexzv**
This combines several PRs.

Should result in functional package data API, which provides per-package details along with fast `/packages/:id/systems` and `/systems/:id/packages` implementations.

Contains:
https://github.com/RedHatInsights/patchman-engine/pull/282
https://github.com/RedHatInsights/patchman-engine/pull/283
https://github.com/RedHatInsights/patchman-engine/pull/284
https://github.com/RedHatInsights/patchman-engine/pull/285

### CODE REVIEW

**Comment by Josca** [comment]
@semtexzv, why did you create this feature branch in origin repo?

**Comment by semtexzv** [comment]
The feature was too big and individual PRs left master in inconsistent state.

File: dev/test_data.sql [hunk 1]
```diff
@@ -33,16 +33,23 @@ INSERT INTO system_platform (id, inventory_id, display_name, rh_account_id,  vma
...
 ('satellite/organization=rh', 1),
 ('satellite/organization=rh', 2),
 ('satellite/organization=ibm', 1);
 
+INSERT INTO package_name(id,name) VALUES
+(1, 'kernel'),
+(2, 'firefox');
+
+INSERT INTO package(id, name_id, version, description, summary) VALUES
+(1, 1, '5.6.13-200.fc31-x86_64', 'kernel 1', 'kernel 1'),
```
[the remark this answers is not in the document]
Comment (semtexzv) [reply dev/test_data.sql:46]: https://github.com/RedHatInsights/patchman-engine/pull/287

File: dev/test_data.sql [hunk 2]
```diff
@@ -33,16 +33,23 @@ INSERT INTO system_platform (id, inventory_id, display_name, rh_account_id,  vma
...
+(2, 2, '76.0.1-1.fc31-x86_64', 'firefox 1', 'firefox 2');
+
+INSERT INTO system_package (system_id, package_id, update_data) VALUES
+(12, 1, '[{"version": "5.10.13-200.fc31-x86_64", "advisory": "RH-100"}]'),
```
[the remark this answers is not in the document]
Comment (semtexzv) [reply dev/test_data.sql:50]: https://github.com/RedHatInsights/patchman-engine/pull/287

File: database_admin/schema/create_schema.sql [hunk 3]
```diff
@@ -535,6 +531,40 @@ CREATE INDEX ON deleted_system (when_deleted);
...
+    id   INT GENERATED BY DEFAULT AS IDENTITY NOT NULL PRIMARY KEY,
+    name TEXT                                 NOT NULL CHECK (NOT empty(name)) UNIQUE
+);
+
+GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE package_name TO vmaas_sync;
+
+CREATE TABLE IF NOT EXISTS package
+(
+    id          INT GENERATED BY DEFAULT AS IDENTITY NOT NULL PRIMARY KEY,
+    -- TODO: inline vs extract ?
```
[the remark this answers is not in the document]
Comment (semtexzv) [reply database_admin/schema/create_schema.sql:545]: Implemented in https://github.com/RedHatInsights/patchman-engine/pull/294

File: vmaas_sync/pkg_sync.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"github.com/RedHatInsights/patchman-clients/vmaas"
+	"github.com/antihax/optional"
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+)
+
+func syncPackages(tx *gorm.DB, pkgs []string) error {
```
Comment (Josca) [inline vmaas_sync/pkg_sync.go:14]: Have you tested it with real vmaas instance? I've got `time="2020-07-27T15:49:59Z" level=error msg="manual called syncing failed" err="Storing packages: Get packages: 400 Bad Request"` after trying to sync packages during perf-testing. I am not able to do perf-testing without synced packages.

Comment (semtexzv) [reply vmaas_sync/pkg_sync.go:14]: Fixes implemented in https://github.com/RedHatInsights/patchman-engine/pull/295

**Pull request merged**